app/pages/Reserved/index.js

```diff
@@ -0,0 +1,379 @@
+    const zone = proof.zones[proof.zones.length - 1];
+
+    for (const key of zone.keys) {
+      if (key.type !== types.DNSKEY)
+        continue;
+
+      const kd = key.data;
+
+      if (!(kd.flags & keyFlags.ZONE))
+        continue;
+
+      if (kd.flags & keyFlags.SEP)
+        continue;
+
+      if (kd.flags & keyFlags.REVOKE)
+        continue;
+
+      key.filename = dnssec.filename(
+        this.state.url,
+        key.data.algorithm,
+        key.data.keyTag()
+      ) + '.private';
+      keys.push(key);
+    }
+
+    this.setState({
+      keys
+    });
+  }
+
+  localSign = async () => {
+    if (!this.state.keys.length || !this.state.proof)
+      return;
+
+    const filenames = [];
+    for (let key of this.state.keys) {
+      filenames.push(key.filename);
+    }
+    const list = filenames.join(', ');
+
+
+    const filePaths = await dialog.showOpenDialog({
+      title: `Open DNSSEC private key file: ${list}`,
+      message: `Open DNSSEC private key file: ${list}`,
+      properties: ["openFile"],
+      filters: {
+        name: "BIND Private Key",
+        extensions: ["private"]
+      }
+    });
+
+    if (!filePaths.filePaths.length)
+      return;
+
+    const filePath = filePaths.filePaths[0];
+
+    let fileName = filePath.split('/');
+
+    fileName = fileName[fileName.length - 1];
+
+    let found = false;
+    for (const acceptable of filenames) {
+      if (fileName === acceptable) {
+        found = true;
+        break;
+      }
+    }
+    if (!found) {
+      this.setState({
+        error: `Can not sign with key ${fileName}`,
+      });
+      return;
+    }
+
+    const keyFile = fs.readFileSync(filePath, 'utf-8');
+    const [alg, privateKey] = dnssec.decodePrivate(keyFile);
+
+    const rr = this.getTXT();
+    const lifespan = 365 * 24 * 60 * 60;
+
+    const sig = dnssec.sign(this.state.keys[0], privateKey, [rr], lifespan);
+
+    const proof = this.state.proof;
+    const zone = proof.zones[proof.zones.length - 1];
+    zone.claim.push(rr);
+    zone.claim.push(sig);
+
+    let json;
+    try {
+      json = await node.sendRawClaim(proof.toBase64());
+    } catch (e) {
+      this.setState({
+        error: `Error sending claim: ${e.message}`,
+      });
+      return;
+    }
+
+    // TODO: success modal with details from json
+  }
+
+  displayKeys() {
+    const rows = [];
+    for (let key of this.state.keys) {
+      const filename = key.filename;
+      key = key.getJSON()
+      rows.push(
+        <tr>
+          <td>{key.data.keyTag}</td>
+          <td>{key.data.keyType}</td>
+          <td>{key.data.algName}</td>
+          <td>{filename}</td>
+        </tr>
+      );
+    }
+
+    return (
+      <div>
+        &nbsp;&nbsp;DNSKEYs available for signing:
+        <table className="reserved__keyList">
+          <tr><td>Key tag</td><td>Key type</td><td>Key Algorithm</td><td>Filename</td></tr>
+          {rows}
+        </table>
+      </div>
+    );
+  }
+
+  render() {
+    const fakesig =
```

Comment:
What do we need this `fakesig` ?

Reply:
oh you got me -- I started working on Method 3 where the user signs completely offline and just pastes their RRSIG into Bob. I had a `textarea` where they paste the sig and this was just for a placeholder. Decided to cut it out for now and push what I had... will pick it up again next week.

Reply:
this is now in use for method 3